src/NaCl.Core/Base/ChaCha20Base.cs

```diff
@@ -2,22 +2,37 @@
 {
     using System;
     using System.Runtime.CompilerServices;
-    using System.Security.Cryptography;
-
     using Internal;
+    using NaCl.Core.Base.ChaChaCore;
 
     /// <summary>
     /// Base class for <seealso cref="NaCl.Core.ChaCha20" /> and <seealso cref="NaCl.Core.XChaCha20" />.
     /// </summary>
     /// <seealso cref="NaCl.Core.Base.Snuffle" />
     public abstract class ChaCha20Base : Snuffle
     {
+        readonly IChaCha20Core _chaCha20Core;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="ChaCha20Base"/> class.
         /// </summary>
         /// <param name="key">The key.</param>
         /// <param name="initialCounter">The initial counter.</param>
-        protected ChaCha20Base(ReadOnlyMemory<byte> key, int initialCounter) : base(key, initialCounter) { }
+        protected ChaCha20Base(ReadOnlyMemory<byte> key, int initialCounter) : base(key, initialCounter)
+        {
+#if INTRINSICS

```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it